packages/models-library/src/models_library/api_schemas_resource_usage_tracker/pricing_plans.py

```diff
@@ -30,30 +31,68 @@ class PricingUnitGet(BaseModel):
                 {
                     "pricing_unit_id": 1,
                     "unit_name": "SMALL",
-                    "unit_extra_info": UnitExtraInfo.model_config["json_schema_extra"]["examples"][0],  # type: ignore [index]
+                    "unit_extra_info": UnitExtraInfoTier.model_config["json_schema_extra"]["examples"][0],  # type: ignore [index]
                     "current_cost_per_unit": 5.7,
                     "current_cost_per_unit_id": 1,
                     "default": True,
                     "specific_info": hw_config_example,
                 }
-                for hw_config_example in HardwareInfo.model_config["json_schema_extra"][
+                for hw_config_example in HardwareInfo.model_config["json_schema_extra"][  # type: ignore[index,union-attr]
                     "examples"
-                ]  # type: ignore[index,union-attr]
+                ]
+            ]
+            + [
+                {
+                    "pricing_unit_id": 2,
+                    "unit_name": "5 seats",
+                    "unit_extra_info": UnitExtraInfoLicense.model_config["json_schema_extra"]["examples"][0],  # type: ignore [index]
+                    "current_cost_per_unit": 10.5,
+                    "current_cost_per_unit_id": 2,
+                    "default": False,
+                    "specific_info": HardwareInfo.model_config["json_schema_extra"][  # type: ignore[index,union-attr]
+                        "examples"
+                    ][
+                        1
+                    ],
+                }
             ]
         }
     )
 
 
-class PricingPlanGet(BaseModel):
+class RutPricingPlanGet(BaseModel):
     pricing_plan_id: PricingPlanId
     display_name: str
     description: str
     classification: PricingPlanClassification
     created_at: datetime
     pricing_plan_key: str
-    pricing_units: list[PricingUnitGet] | None
+    pricing_units: list[RutPricingUnitGet] | None
     is_active: bool
 
+    @model_validator(mode="after")
+    def ensure_classification_matches_extra_info(self):
+        """Enforce that all PricingUnitGet.unit_extra_info match the plan's classification."""
+        if not self.pricing_units:
+            return self  # No units to check
+
+        for unit in self.pricing_units:
+            if (
+                self.classification == PricingPlanClassification.TIER
```

Comment:
FYI: This is a classic example that can be solved with discriminator fields

SEE https://docs.pydantic.dev/latest/concepts/unions/